src/ObjectDetector/index.js

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2020-2024 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+/*
+ * ObjectDetector: Object detection using cocoSsd in the browser
+ */
+
+import * as tf from "@tensorflow/tfjs";
+import * as cocoSsd from "@tensorflow-models/coco-ssd";
+import callCallback from "../utils/callcallback";
+import handleArguments from "../utils/handleArguments";
+import handleOptions from "../utils/handleOptions";
+import { handleModelName } from "../utils/handleOptions";
+import { mediaReady } from "../utils/imageUtilities";
+
+class ObjectDetector {
+  /**
+   * An object for configuring ObjectDetector options.
+   * @typedef {Object} configOptions
+   * @property {string} modelType - Optional. The type of model to use. Default: "lite".
+   */
+
+  /**
+   * Creates ObjectDetector.
+   * @param {configOptions} options - An object containing ObjectDetector configuration options.
+   * @param {function} callback - A callback to be called when the model is ready.
+   * @private
+   */
+  constructor(modelName, options, callback) {
+    this.modelName = handleModelName(modelName, ["cocoSsd"], "cocoSsd", "objectDetector");
+    this.model = null;
+    this.config = options;
+    this.runtimeConfig = {};
+    this.detectMedia = null;
+    this.detectCallback = null;
+
+    // flags for detectStart() and detectStop()
+    this.detecting = false; // True when detection loop is running
+    this.signalStop = false; // Signal to stop the loop
+    this.prevCall = ""; // Track previous call to detectStart() or detectStop()
+
+    this.ready = callCallback(this.loadModel(), callback);
+  }
+
+  /**
+   * Loads the model.
+   * @return {this} the ObjectDetector model.
+   * @private
+   */
+  async loadModel() {
+    await tf.ready();
+    this.model = await cocoSsd.load();
+    console.log('Finished loading cocoSsd');
+    return this;
+  }
+
+  /**
+   * Asynchronously outputs a single object detection result when called.
+   * Supports both callback and promise.
+   * @param {*} [media] - An HMTL or p5.js image, video, or canvas element to run the detection on.
+   * @param {function} [callback] - Optional. A callback to handle the detection result.
+   * @returns {Promise<Array>} The detection result.
+   */
+  async detect(...inputs) {
+    const argumentObject = handleArguments(...inputs);
+    argumentObject.require("image", "An html or p5.js image, video, or canvas element argument is required for detect().");
+    const { image, callback } = argumentObject;
+
+    await mediaReady(image, false);
+
+    const predictions = await this.model.detect(image);
+    console.log('raw result from cocoSsd', predictions);
+
+    const result = predictions;
```

Comment:
Perhaps worth having a look at the previous version of ml5 to see in which format it returned data to the user. Not that you necessarily need to imitate it, but not so sure about `bbox`..!

From a quick look, previously the result was an array of objects, each with the following properties:
```
label: String,
confidence: Number,
x: Number (px),
y: Number (px),
width: Number (px),
height: Number (px),
normalized: {
  x: Number (0-1),
  y: Number (0-1),
  width: Number (0-1),
  height: Number (0-1),
}
```

(the `normalized` part is perhaps less common in ml5, and perhaps we could drop this - but the rest seems to be similar to e.g. `faceMesh` returns results presently)